docs/source/developer/patterns.rst

```diff
@@ -121,7 +121,7 @@ Data Structures
 ---------------
 
 For public API, we have three options: JavaScript ``Array``,
-``IIterator``, and ``ReadonlyArray`` (an interface defined by
+``Iterator``, and ``ReadonlyArray`` (an interface defined by
```

Comment:
`IIterator` is not a typo; the initial `I` indicates that this is an interface.

Reply:
Had mixed feelings about those, thanks!